codeU4.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Jun 10 09:17:26 2017
+Exercise 4, version 1
+@author: Luiza
+"""
+
+import unittest
+import numpy as np
+
+class Tile(object):
+
+    '''Given (n by m) bool 2D array consisting of islands as True
+    and water as False
+    Counts the number of separate islands, that can be formed using 4-connectivity
+    between the land tiles
+    '''
+
+    def __init__(self, bool_matrix, rows, cols):
+
+        '''Initialization of tile as 2D numpy bool array (True/False or 0/1)
+        checks whether we have the correct size and expects boolean array elements
+
+        Args:
+            bool_matrix: boolean numpy 2D array for tile
+            rows, cols: size of the tile array
+        '''
+
+        (self.rows, self.cols) = np.shape(bool_matrix)
+        self.tile = bool_matrix
+        assert self.rows == rows or self.cols == cols, \
+        "Given matrix dimensions mismatch between the actual size of array"
+        self.num_islands = 0
+        self.unvisited_land_cells = set((i, j) for i in range(self.rows) \
+        for j in range(self.cols) if self.tile[i, j])
+
+    def is_unvisited_island(self, tuple_cell):
+
+        """Checks if a given cell (x,y) is a land and was not visited so far
+
+        Args:
+            tuple_cell: a tuple (x,y) coordinate
+
+        Returns:
+            True: if not visited land
+            False: otherwise
+        """
+
+        return (0 <= tuple_cell[0] < self.rows and 0 <= tuple_cell[1] < self.cols \
+        and (tuple_cell[0], tuple_cell[1]) in self.unvisited_land_cells)
+
```

Comment:
Again, I really like your way of handling the checking of cells! And I really like the list comprehension! 
One minor thing, I think you don't need to unpack the **tuple_cell** ? The following seems to work equally well: 
```python
return ((0, 0) <= tuple_cell < self.unvisited_land_cells 
        and tuple_cell in self.unvisited_land_cells)
```